app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/interactors/Interactor.kt

```diff
@@ -0,0 +1,7 @@
+package de.tum.`in`.tumcampusapp.component.ui.cafeteria.interactors
+
+interface Interactor<T> {
+
+    fun execute(): T
```

Comment:
> encapsulate a use case.

I'm not sure this helps... I still don't get the difference to a `Callable` except a fancier name.

Nevertheless, please document this *in the code* and provide a usage example.

> we can easily inject it via the constructor instead of having duplicate code

I really don't see this behavior in this interface :man_shrugging: